[BUG] Panics when attempting to fetch a chart from an OCI registry as a dependency of a local chart

**Describe the bug**
When attempting to template a local chart that has a dependency to a chart that is stored within an OCI registry, khelm panics:
```
Running khelm 2.3.3 (helm 3.11.3)
Building/fetching chart REDACTED 0.0.0 dependencies
Saving 1 charts
Downloading platform-service from repo oci://REDACTED
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x15787cd]

goroutine 1 [running]:
helm.sh/helm/v3/pkg/registry.(*Client).Pull(0x0, {0xc007c9dc26?, 0xc000558e48?}, {0x0, 0x0, 0x3?})
	/home/runner/go/pkg/mod/helm.sh/helm/v3@v3.11.3/pkg/registry/client.go:303 +0x42d
helm.sh/helm/v3/pkg/getter.(*OCIGetter).get(0xc000558e68?, {0xc007c9dc20, 0x4d})
	/home/runner/go/pkg/mod/helm.sh/helm/v3@v3.11.3/pkg/getter/ocigetter.go:53 +0x274
helm.sh/helm/v3/pkg/getter.(*OCIGetter).Get(0xc0001b7290, {0xc007c9dc20, 0x4d}, {0xc007cd35c0, 0x6, 0x4?})
	/home/runner/go/pkg/mod/helm.sh/helm/v3@v3.11.3/pkg/getter/ocigetter.go:36 +0x6f
helm.sh/helm/v3/pkg/downloader.(*ChartDownloader).DownloadTo(0xc000559290, {0xc007cb3db0, 0x45}, {0xc007cb3df6?, 0x1?}, {0xc007d0e080, 0x7b})
	/home/runner/go/pkg/mod/helm.sh/helm/v3@v3.11.3/pkg/downloader/chart_downloader.go:100 +0xea
helm.sh/helm/v3/pkg/downloader.(*Manager).downloadAll(0xc000738240, {0xc000716510?, 0x1, 0x1})
	/home/runner/go/pkg/mod/helm.sh/helm/v3@v3.11.3/pkg/downloader/manager.go:356 +0xeba
helm.sh/helm/v3/pkg/downloader.(*Manager).Update(0xc000738240)
	/home/runner/go/pkg/mod/helm.sh/helm/v3@v3.11.3/pkg/downloader/manager.go:201 +0x109
helm.sh/helm/v3/pkg/downloader.(*Manager).Build(0xc000738240)
	/home/runner/go/pkg/mod/helm.sh/helm/v3@v3.11.3/pkg/downloader/manager.go:96 +0x3c6
github.com/mgoltzsche/khelm/v2/pkg/helm.buildChartDependencies({0x1d00f40, 0xc000461b40}, 0xc000163b00?, {0xc0000ef400, 0x71}, 0xc00089cd80, {0x1d07440, 0xc000163b00}, 0xc000947870, {0xc000721f00, ...})
	/home/runner/work/khelm/khelm/pkg/helm/load.go:252 +0x2a9
github.com/mgoltzsche/khelm/v2/pkg/helm.buildLocalCharts({0x1d00f40, 0xc000461b40}, {0xc0008d0460?, 0x1, 0x4?}, 0xc00089cd80, {0x1d07440, 0xc000163b00}, 0xc0000061a0?, {0xc000721f00, ...})
	/home/runner/work/khelm/khelm/pkg/helm/load.go:203 +0x29d
github.com/mgoltzsche/khelm/v2/pkg/helm.(*Helm).buildAndLoadLocalChart(0xc00089cc60, {0x1d00f40, 0xc000461b40}, 0xc00089cd80)
	/home/runner/work/khelm/khelm/pkg/helm/load.go:114 +0x425
github.com/mgoltzsche/khelm/v2/pkg/helm.(*Helm).loadChart(0xf00000090?, {0x1d00f40, 0xc000461b40}, 0xc00089cd80)
	/home/runner/work/khelm/khelm/pkg/helm/load.go:32 +0x2e5
github.com/mgoltzsche/khelm/v2/pkg/helm.(*Helm).Render(0xc00089cc60, {0x1d00f40, 0xc000461b40}, 0xc00089cd80)
	/home/runner/work/khelm/khelm/pkg/helm/render.go:39 +0x42e
main.render(0x1a2c82d?, 0x1?)
	/home/runner/work/khelm/khelm/cmd/khelm/common.go:25 +0x134
main.templateCommand.func2(0xc0008a2300?, {0xc00002a260, 0x1, 0x1?})
	/home/runner/work/khelm/khelm/cmd/khelm/template.go:49 +0x1e5
github.com/spf13/cobra.(*Command).execute(0xc0008a2300, {0xc00002a230, 0x1, 0x1})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc0008a2000)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
main.Execute({0x1cea2e0?, 0xc000012010}, {0x1cea340?, 0xc000012018})
	/home/runner/work/khelm/khelm/cmd/khelm/root.go:95 +0x885
main.main()
	/home/runner/work/khelm/khelm/cmd/khelm/main.go:9 +0x35
```

**To Reproduce**
```sh
khelm template ./charts/my-local-chart-that-has-an-oci-dependency
```
(I cannot provide more information since it is a proprietary project.)

**Expected behavior**
khelm should template a chart that has a dependency to a chart that lives within an OCI registry.

**khelm version:**
khelm 2.3.3 (helm 3.11.3)

--- a/example/oci-dependency/Chart.lock
+++ b/example/oci-dependency/Chart.lock
@@ -1,0 +1,6 @@
+dependencies:
+- name: karpenter
+  repository: oci://public.ecr.aws/karpenter
+  version: 0.35.0
+digest: sha256:c351689f0d2d254c02b7e8aac54c06fb66e8871dfae3479cfbf1af88e1fee268
+generated: "2024-04-11T01:47:48.41682948+02:00"

--- a/example/oci-dependency/Chart.yaml
+++ b/example/oci-dependency/Chart.yaml
@@ -1,0 +1,8 @@
+apiVersion: v2
+description: example chart with OCI chart as dependency
+name: oci-dependency
+version: 0.0.0
+dependencies:
+- name: karpenter
+  version: "0.35.0"
+  repository: "oci://public.ecr.aws/karpenter"

--- a/example/oci-dependency/generator.yaml
+++ b/example/oci-dependency/generator.yaml
@@ -1,0 +1,6 @@
+apiVersion: khelm.mgoltzsche.github.com/v2
+kind: ChartRenderer
+metadata:
+  name: oci-dependency
+  namespace: kube-system
+chart: .

--- a/example/oci-image/generator.yaml
+++ b/example/oci-image/generator.yaml
@@ -3,5 +3,6 @@ kind: ChartRenderer
 metadata:
   name: karpenter
   namespace: kube-system
-chart: "oci://public.ecr.aws/karpenter/karpenter"
+chart: karpenter
 version: 0.35.0
+repository: "oci://public.ecr.aws/karpenter"

--- a/pkg/helm/load.go
+++ b/pkg/helm/load.go
@@ -230,12 +230,19 @@ func buildLocalCharts(ctx context.Context, localCharts []localChart, cfg *config
 }
 
 func buildChartDependencies(ctx context.Context, chartRequested *chart.Chart, chartPath string, cfg *config.LoaderConfig, repos repositoryConfig, settings *cli.EnvSettings, getters getter.Providers) error {
+	registryClient, err := registry.NewClient(
+		registry.ClientOptEnableCache(true),
+	)
+	if err != nil {
+		return errors.WithStack(err)
+	}
 	man := &downloader.Manager{
 		Out:              log.Writer(),
 		ChartPath:        chartPath,
 		Keyring:          cfg.Keyring,
 		SkipUpdate:       true,
 		Getters:          getters,
+		RegistryClient:   registryClient,
 		RepositoryConfig: settings.RepositoryConfig,
 		RepositoryCache:  settings.RepositoryCache,
 		Debug:            settings.Debug,
@@ -244,7 +251,7 @@ func buildChartDependencies(ctx context.Context, chartRequested *chart.Chart, ch
 		man.Verify = downloader.VerifyAlways
 	}
 	// Workaround for leftover tmpcharts dir (which makes Build() fail)
-	err := os.RemoveAll(filepath.Join(chartPath, "tmpcharts"))
+	err = os.RemoveAll(filepath.Join(chartPath, "tmpcharts"))
 	if err != nil {
 		return errors.WithStack(err)
 	}

--- a/pkg/helm/load.go
+++ b/pkg/helm/load.go
@@ -38,6 +38,10 @@ func (h *Helm) loadChart(ctx context.Context, cfg *config.ChartConfig) (*chart.C
 		} else {
 			return nil, errors.Errorf("chart directory %q not found and no repository specified", cfg.Chart)
 		}
+	} else if registry.IsOCI(cfg.Repository) {
+		cfg.Chart = fmt.Sprintf("%s/%s", cfg.Repository, cfg.Chart)
+		cfg.Repository = ""
+		return h.loadOCIChart(ctx, cfg)
 	}
 	return h.loadRemoteChart(ctx, cfg)
 }

--- a/pkg/helm/render_test.go
+++ b/pkg/helm/render_test.go
@@ -55,6 +55,8 @@ func TestRender(t *testing.T) {
 		{"include", "example/include/generator.yaml", []string{}, "  key: b", nil},
 		{"local-chart-with-local-dependency-and-transitive-remote", "example/localrefref/generator.yaml", []string{}, "rook-ceph-v0.9.3", nil},
 		{"local-chart-with-remote-dependency", "example/localref/generator.yaml", []string{}, "rook-ceph-v0.9.3", nil},
+		{"oci-chart", "example/oci-image/generator.yaml", []string{"kube-system", "kube-node-lease"}, "name: ec2nodeclasses.karpenter.k8s.aws", nil},
+		{"oci-dependency", "example/oci-dependency/generator.yaml", []string{"kube-system", "kube-node-lease"}, "name: ec2nodeclasses.karpenter.k8s.aws", nil},
 		{"values-inheritance", "example/values-inheritance/generator.yaml", []string{}, " inherited: inherited value\n  fileoverwrite: overwritten by file\n  valueoverwrite: overwritten by generator config", nil},
 		{"cluster-scoped", "example/cluster-scoped/generator.yaml", []string{}, "myrolebinding", nil},
 		{"chart-hooks", "example/chart-hooks/generator.yaml", []string{"default"}, "  key: myvalue", []string{

